### ISSUE TITLE
Safari 9: inserting into a different document fires inserted 

### ISSUE THREAD

**Issue by daffl** [canjs/can-util#86]
https://github.com/canjs/can-util/blob/master/dom/mutate/mutate-test.js#L36 is failing on Safari 9.0:

![screen shot 2016-09-20 at 4 34 40 pm](https://cloud.githubusercontent.com/assets/338316/18692595/36996360-7f50-11e6-8969-41e1362acd7e.png)

**Comment by matthewp** [comment]
I think this is related to what you were seeing yesterday, with Safari not firing events unless it's in the DOM.

**Comment by matthewp** [comment]
This doesn't work in Safari:

``` html
<script>
(function(){
    var doc = document.implementation.createHTMLDocument('Demo');

    var div = document.createElement("div");
    div.addEventListener("inserted", function(){
        console.log("called");
    });

  doc.body.appendChild(div);

  var ev = doc.createEvent('HTMLEvents');
  ev.initEvent("inserted", true, false);
  div.dispatchEvent(ev);
})();
</script>
```

Nor does `new Event()`

**Comment by matthewp** [comment]
Submitted this: https://github.com/w3c/web-platform-tests/pull/3788

If they don't object I'll also file a bug against WebKit.

Thinking about a workaround.

**Comment by matthewp** [comment]
So what this test is really just testing is that mutate uses the `DOCUMENT()` getter to know which document to fire the `inserted` event against. I'm not sure how else to test this.

**Comment by daffl** [comment]
If it is a bug in Webkit we can probably just run the test conditionally for now?

**Comment by matthewp** [comment]
I guess we could. I'm going to wait and see what they say about the test I submitted and come back to this later.

**Comment by daffl** [comment]
I created #93 which feature detects if the test can be run so that we'll at least get everything else to pass on Safari.

**Comment by matthewp** [comment]
Fixed by #93
